### PR TITLE
agent: Rework protocol generation script

### PR DESCRIPTION
**Pull request by jodh-intel**
Major overhaul of the protocol generation script to make it clearer (and
thus easier to maintain) and more consistent with our other scripts.

Significant other changes:

- Updated the URLs so that the protocol files are read from the Kata
  2.x repo rather than the 1.x one now 2.0 has been released.

- Removed golang support as this is no longer required for Kata 2.x.

Fixes: #1015.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
To test it:

```bash
$ cd $GOPATH/src/github.com/kata-containers/kata-containers/src/agent
$ ./protocols/hack/update-generated-proto.sh all && make &>/dev/null && echo OK
INFO: Converting 'agent.proto' into 'agent.rs'
INFO: Converting 'health.proto' into 'health.rs'
INFO: Converting 'oci.proto' into 'oci.rs'
github.com/kata-containers/kata-containers/src/agent/protocols/protos/oci.proto:15:1: warning: Import google/protobuf/wrappers.proto is unused.
INFO: Converting 'github.com/kata-containers/agent/pkg/types/types.proto' into 'types.rs'
INFO: Done
OK
```

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jodh-intel!

File: src/agent/protocols/hack/update-generated-proto.sh [hunk 1]
```diff
@@ -1,142 +1,188 @@
 #!/bin/bash
 
-# //
-# // Copyright 2020 Ant Financial
-# //
-# // SPDX-License-Identifier: Apache-2.0
-# //
+# Copyright 2020 Ant Financial
```
Comment (devimc) [inline src/agent/protocols/hack/update-generated-proto.sh:3]: might be you should update the `Copyright` too :smile:

**Review by devimc — APPROVED** [review]
thanks @jodh-intel - lgtm - one nit

File: src/agent/protocols/hack/update-generated-proto.sh (hunk 1 above)
Comment (jodh-intel) [reply src/agent/protocols/hack/update-generated-proto.sh:3]: Done ;)

Comment (devimc) [reply src/agent/protocols/hack/update-generated-proto.sh:3]: thanks

File: src/agent/protocols/hack/update-generated-proto.sh [hunk 2]
```diff
@@ -1,142 +1,184 @@
...
+
+	[ -e "$filepath" ] || die "file $file ($filepath) does not exist"
+
+	local out_dir="./protocols/src"
+	local out_file=$(basename "${file/.proto/.rs}")
+	local outfile_path="${out_dir}/${out_file}"
+
+	local ttrpc_cmd=$(command -v ttrpc_rust_plugin)
+
+	local includes=()
```
Comment (jodh-intel) [inline src/agent/protocols/hack/update-generated-proto.sh:102]: I just noticed some remaining 1.x URLs here. ooi, I commented them out and *all* of them are redundant! I've left in only the essential includes.

Branch updated :smile:

**Comment by Tim-Zhang** [comment]
@jodh-intel According your reply https://github.com/kata-containers/kata-containers/pull/1021#pullrequestreview-514427498, I close it now.

**Comment by jodh-intel** [comment]
Thanks @Tim-Zhang ! ;)

**Pull request closed without merging**